be/src/vec/functions/function_string.h

```diff
@@ -3770,9 +3770,10 @@ class FunctionConvertTo : public IFunction {
         auto& res_offset = col_res->get_offsets();
         auto& res_chars = col_res->get_chars();
         res_offset.resize(input_rows_count);
-        // max pinyin size is 6, double of utf8 chinese word 3, add one char to set '~'
-        ColumnString::check_chars_length(str_chars.size() * 2 + input_rows_count, 0);
-        res_chars.resize(str_chars.size() * 2 + input_rows_count);
+        // max pinyin size is 6 + 1 (first '~') for utf8 chinese word 3
+        size_t pinyin_size = (str_chars.size() + 2) / 3 * 7;
```

Comment:
this size +2 means?
maybe could add some comment for the get the pinyin_size

Reply:
pinyin_size = chars_size / 3 * 7

but we know 1/3=0, 2/3=0, so we need: (chars_size + (3-1)) / 3 * 7